feat: reimplement #[type = ..] on enums for `enum` generation

pintariching opened the pull request:
This is a reimplementation of #24. I tried merging that PR into main but I had some merge issues because the PR is old. 
All the tests pass and you can now add `#[ts(type = "enum"]` or `#[ts(type = "const enum"]` to generate an enum like 
```rust
#[ts(type = "const enum")]
enum Dog {
  Woof,
  Bark,
  Howl
}
```
generates a typescript enum like 
```typescript
const enum Dog = { Woof, Bark, Howl };
```
Credits to @JakubKoralewski for implementing this.

NyxCode commented:
Thanks, really appreciate it! The tests currently seem to fail, do you know why?

pintariching commented:
The problem seems to be running `cargo test --no-default-features`. I'll check it out.

pintariching commented:
Ok I think I fixed it. The tests were using attribute macros from `serde`like `serde(rename_all = ...)` instead of `ts(rename_all = ...)` without having having the right directives enabled. I changed them to the latter now it should pass.

lorenzolewis commented:
@NyxCode is there any possibility for this one to be merged in when you have capacity?

escritorio-gustavo commented:
@NyxCode  This PR hasn't been active since 2022. Considering that and the fact that TS enums are generally considered to be really bad, should this PR be closed?

NyxCode commented:
@escritorio-gustavo Agreed, especially with your current work on enums in mind, which might require some deep refactors.  
If @pintariching is still interested in this, it'd probably make sense to re-write this based on the current master once enum flattening lands.

escritorio-gustavo commented:
Great, given that I'll also close #24, since this is a reimplementation of that
